packages/outbox-core/lib/accumulators.ts

```diff
@@ -0,0 +1,73 @@
+import type { CommonEventDefinition } from '@message-queue-toolkit/schemas'
+import type { OutboxEntry } from './objects.ts'
+
+/**
+ * Accumulator is responsible for storing outbox entries in two cases:
+ * - successfully dispatched event
+ * - failed events
+ *
+ * Thanks to this, we can use aggregated result and persist in the storage in batches.
+ */
+export interface OutboxAccumulator<SupportedEvents extends CommonEventDefinition[]> {
+  /**
+   * Accumulates successfully dispatched event.
+   * @param outboxEntry
+   */
+  add(outboxEntry: OutboxEntry<SupportedEvents[number]>): Promise<void>
+
+  /**
+   * Accumulates failed event.
+   * @param outboxEntry
+   */
+  addFailure(outboxEntry: OutboxEntry<SupportedEvents[number]>): Promise<void>
+
+  /**
+   * It's meant to be used by OutboxStorage::flush() to get all entries that should be persisted as successful ones.
```

Comment:
I would rather document what it does (return all entries accumulated so far) rather than document who and when is expected to use it, as these may vary. E. g. we already use it for prefiltering as well

Reply:
Sure, updated.